### PR TITLE
Fix: Broken urls and branch names in readme

### PR DESCRIPTION
**Pull request by deepakchethan**
Hello!

I was going through the readme and noticed that some urls are outdated and branch name was still pointing to the old `master` instead of `main` and updated them. Please see if we can update these if it's fine.

Thanks!

### CODE REVIEW

**Review by Tymek — APPROVED** [review]
Thx @deepakchethan! I think `.github/workflows/main.yml` and `.github/workflows/publish_javadoc.yml` can also be updated by removing reference to unused branch.

**Review by ivarconr — APPROVED** [review]

**Comment by deepakchethan** [comment]
> Thx @deepakchethan! I think `.github/workflows/main.yml` and `.github/workflows/publish_javadoc.yml` can also be updated by removing reference to unused branch.

Sure, Thank you for the review. I updated them in https://github.com/Unleash/unleash-client-java/pull/155/commits/d7d6ae78b22fc2d695141439ca6a2f6b3500b022.

File: README.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ provided by unleash.
 - RemoteAddressStrategy
 - ApplicationHostnameStrategy
 
-Read more about the strategies in [activation-strategy.md](https://github.com/Unleash/unleash/blob/master/docs/activation-strategies.md).
+Read more about the strategies in [activation-strategy.md](https://github.com/Unleash/unleash/blob/main/website/docs/user_guide/activation-strategies.md).
```
Comment (thomasheartman) [inline README.md:78]:
```suggestion
Read more about the strategies in [the activation strategies documentation](https://docs.getunleash.io/user_guide/activation_strategy).
```

File: README.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ Unleash unleash return new DefaultUnleash(config, s1, s2);
 
 ### Unleash context
 
-In order to use some of the common activation strategies you must provide a [unleash-context](https://github.com/Unleash/unleash/blob/master/docs/unleash-context.md).
+In order to use some of the common activation strategies you must provide a [unleash-context](https://github.com/Unleash/unleash/blob/main/website/docs/user_guide/unleash-context.md).
```
Comment (thomasheartman) [inline README.md:93]:
```suggestion
In order to use some of the common activation strategies you must provide an [Unleash context](https://docs.getunleash.io/user_guide/unleash_context).
```

**Review by thomasheartman — CHANGES_REQUESTED** [review]
Super work! Thanks for picking up on this 🙏🏼  I've added a couple of inline suggestions of link updates, so that they point to the actual rendered docs instead of to the github markdown source files. Once that's in, it's all good from my end 😄

**Pull request merged**